dhall/tests/Dhall/Test/QuickCheck.hs

```diff
@@ -500,14 +509,20 @@ instance Arbitrary URL where
     arbitrary = do
         scheme <- arbitrary
 
-        authority <- arbitrary
+        -- TODO: the authority generator could be more precise, but it currently
+        -- seems good enough.
+        let validAuthorityChar =
+                arbitrary `suchThat` \ch ->
+                    not (Data.Char.isSpace ch) && not (Data.Char.isPunctuation ch)
+
+        authority <- Text.pack <$> Test.QuickCheck.listOf validAuthorityChar
 
         let validPChar =
                 Test.QuickCheck.frequency
                     [ (26, chooseCharacter ('\x41', '\x5A'))
                     , (26, chooseCharacter ('\x61', '\x7A'))
                     , (10, chooseCharacter ('\x30', '\x39'))
```

Comment:
Is there a reason for using explicit Unicode instead of the following?
```suggestion
                    [ (26, chooseCharacter ('A', 'Z'))
                    , (26, chooseCharacter ('a', 'z'))
                    , (10, chooseCharacter ('0', '9'))
```

Reply:
> Is there a reason for using explicit Unicode instead of the following?

Maybe it's about correspondence to the grammar (which AFAIK uses the hex codes because character literals are case-insensitive in ABNF)? https://github.com/dhall-lang/dhall-lang/blob/84b16fd73cdd4a15bffb29af83c7815675ee57a7/standard/dhall.abnf#L183-L189

Honestly I had no idea that these are the `ALPHANUM` chars before you pointed it out though! ;)

@Gabriel439 What do you think?

Reply:
Yeah, it's fine to use the actual characters.  I only used the escape codes because I was too lazy to lookup the matching characters